components/org.wso2.carbon.identity.scim.common/src/main/java/org/wso2/carbon/identity/scim/common/utils/SCIMCommonUtils.java

```diff
@@ -217,6 +222,43 @@ public static void setAdminSCIMAttributes() {
         }
     }
 
+    public static void addAdminGroup(UserStoreManager userStoreManager) throws UserStoreException {
+        try {
+            SCIMGroupHandler scimGroupHandler = new SCIMGroupHandler(userStoreManager.getTenantId());
+
+            String domainName = UserCoreUtil.getDomainName(userStoreManager.getRealmConfiguration());
+            if (domainName == null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Domain name is null and setting default domain as " + UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME);
+                }
+                domainName = UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME;
```

Comment:
Use utils to get Primary domain name

Reply:
Since primary domain can be changed to something else than "Primary"

Reply:
There is no util in the user core to get the primary domain.

Reply:
Sorry, There is a method in identity utils.